app/src/main/java/com/example/checkpoint/dto/Weather.kt

```diff
@@ -5,4 +5,9 @@ data class Weather (
     val main: String,
     val description: String,
     val icon: String
-)
\ No newline at end of file
+){
+    private var CurrentWeather = "$main $id $description $icon"
+    override fun toString() : String{
+        return CurrentWeather
```

Comment:
No need to use a separate variable here.  Just have this return "$main $id $description $icon" directly.